src/libra_py/cube_file_process.py

```diff
@@ -0,0 +1,255 @@
+
+    """
+
+    f = open(filename,'r')
+    lines = f.readlines()
+    f.close()
+	
+	# The number of atoms in the 3rd line
+    natoms = int(lines[2].split()[0])
+	
+	# The number of voxels defined for each axis
+    vx = int(lines[3].split()[0])
+    vy = int(lines[4].split()[0])
+    vz = int(lines[5].split()[0])
+	
+	# The spacing vector. This will be used in the 'marching_cubes_lewiner'
+	# function to define the vertices and faces for 'plot_trisurf' function.
+    spacing = []
+    spacing.append([float(lines[3].split()[1])+float(lines[4].split()[1])+float(lines[5].split()[1]),\
+               float(lines[3].split()[2])+float(lines[4].split()[2])+float(lines[5].split()[2]),\
+               float(lines[3].split()[3])+float(lines[4].split()[3])+float(lines[5].split()[3])])
+    
+    # Bohr to Angstrom : *0.529177
+	# Here we use the same unit as is used in the .cube file structure which is Bohr.
+	# The three vectors below are the same vectors which are present in lines 4th to 6th
+	# which are used to create the 3D numpy array of the grid points (x, y, z) which are
+	# used to plot the isosurfaces.
+    vecx = np.array([float(lines[3].split()[1]),float(lines[3].split()[2]),float(lines[3].split()[3])])#*0.529177
+    vecy = np.array([float(lines[4].split()[1]),float(lines[4].split()[2]),float(lines[4].split()[3])])#*0.529177
+    vecz = np.array([float(lines[5].split()[1]),float(lines[5].split()[2]),float(lines[5].split()[3])])#*0.529177
+
+    # First we read all the isovalues in a 1D array
+    s = []
+
+    for i in range(natoms+3+2+1,len(lines)):
+        for j in range(0,len(lines[i].split())):
+            s.append(float(lines[i].split()[j]))
+
+    v = np.zeros((vx,vy,vz)) # Define the volumetric numpy matrix
+    
+	# Setting up the counters
+    c = 0
+    c1 = 0
+    c2 = 0
+	
+    for i in range(0,len(s)):
+        if c2!=vx:
+            v[c2][c1][c] = s[i]
+            c = c+1
+            if c%vz==0:
+                c = 0
+                c1 = c1+1
+                if c1%vy==0:
+                    c1 = 0
+                    c2 = c2+1
+    
+	# Now define the x, y, and z 3D arrays to store the grids
+	# which is then used for plotting the isosurfaces.
+    x = np.zeros((vx,vy,vz))
+    y = np.zeros((vx,vy,vz))
+    z = np.zeros((vx,vy,vz))
+	
+	# Defining each element of the grid points.
+    for i in range(0,vx):
+        for j in range(0,vy):
+            for k in range(0,vz):
+                x[i][j][k] = vecx[0]*i+vecy[0]*j+vecz[0]*k
+                y[i][j][k] = vecx[1]*i+vecy[1]*j+vecz[1]*k
+                z[i][j][k] = vecx[2]*i+vecy[2]*j+vecz[2]*k
+
+    # For plottin the atoms in the molecule we have to read the
+	# coordinates in xyz format which starts from the 7th line.
+    coordinates = []
+    for i in range(6,natoms+6):
+    	coordinates.append(lines[i].split())
+
+    coordinates = np.array(coordinates)
+                
+    return coordinates, x, y, z, v, spacing
+
+
+
+
+def cube_ovlp_arb(A, B, dv):

```

Comment:
* the names of the arguments are too short, not in compliance with PEP8
* the name of the function could be more descriptive